lib/galaxy/tools/toolbox/base.py

```diff
@@ -1217,9 +1217,10 @@ def _init_dependency_manager(self):
             return
         app_config_dict = self.app.config.config_dict
         conf_file = app_config_dict.get("dependency_resolvers_config_file")
-        default_tool_dependency_dir = os.path.join(self.app.config.data_dir, self.app.config.appschema['tool_dependency_dir'].get('default'))
-        self.dependency_manager = build_dependency_manager(app_config_dict=app_config_dict, conf_file=conf_file,
-                                                           default_tool_dependency_dir=default_tool_dependency_dir)
+        default_tool_dependency_dir = os.path.join(
+            self.app.config.data_dir, self.app.config.schema.app_schema['tool_dependency_dir'].get('default'))
```

Comment:
What is the distinction between `config.schema.defaults['foo']` and `config.schema.app_schema['foo'].get('default')`?

Reply:
@natefoo None at all! It makes sense to use it, of course; but the mock schema has no property `defaults` - so toolbox unit tests will fail. It's a trade-off between adding some minor technical dept here, and modifying galaxy mock (`MockAppConfig`, which has a definition for `MockSchema`, which is a `Bunch` and doesn't have a `defaults` property). etc.

Reply:
Ah ok. It'd be fairly difficult for folks to know whether it's safe to use `config.schema.defaults` in a given scenario then, though?

Reply:
Yes, indeed.. It's supposed to simplify, not to add a special case. I'll look into modifying the schema mock object to accommodate this. Leaving this open/unresolved for now.

Reply:
This, too, proved to be annoyingly simple. See last commit.